File: android/app/src/main/java/io/raindrop/raindropio/NativeBridge/NativeBridge.java
package io.raindrop.raindropio.NativeBridge;

import android.app.Activity;
import android.content.ContentResolver;
import android.content.Intent;
import android.content.res.TypedArray;

import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.content.res.Configuration;
import android.net.Uri;
import android.os.Build;
import android.view.View;
import android.view.Window;
import android.webkit.MimeTypeMap;

import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

import java.util.HashMap;
import java.util.Map;

import static com.facebook.react.bridge.UiThreadUtil.runOnUiThread;

public class NativeBridge extends ReactContextBaseJavaModule {
    ReactApplicationContext context;

    public NativeBridge(ReactApplicationContext reactContext) {
        super(reactContext);
        context = reactContext;
    }

    @Override
    public String getName() {
        return "NativeBridge";
    }

    @Override
    public Map<String, Object> getConstants() {
        final Map<String, Object> constants = new HashMap<>();

        //Is Tablet?
        constants.put("isTablet", false);
        try {
            int layout = this.context.getResources().getConfiguration().screenLayout & Configuration.SCREENLAYOUT_SIZE_MASK;
            constants.put("isTablet", layout == Configuration.SCREENLAYOUT_SIZE_LARGE || layout == Configuration.SCREENLAYOUT_SIZE_XLARGE);
        } catch (Exception e) {}

        //Version
        constants.put("appVersion", "0.0.0");
        try {
            final PackageManager packageManager = this.context.getPackageManager();
            final String packageName = this.context.getPackageName();
            PackageInfo info = packageManager.getPackageInfo(packageName, 0);
            constants.put("appVersion", info.versionName);
        } catch (Exception e) {}

        return constants;
    }

    @ReactMethod
    public void openFileUrl(String url, String mimeType) {
        ContentResolver cR = context.getContentResolver();
        MimeTypeMap mime = MimeTypeMap.getSingleton();

        //details
        Uri uri = Uri.parse(url);
        String detectedMimeType = mime.getExtensionFromMimeType(cR.getType(uri));

        //start activity
        Intent intent = new Intent(Intent.ACTION_VIEW);
        intent.setDataAndType(uri, (mimeType != null && !mimeType.isEmpty()) ? mimeType : detectedMimeType);
        intent.setFlags(Intent.FLAG_ACTIVITY_NO_HISTORY);
        getCurrentActivity().startActivity(Intent.createChooser(intent, ""));
    }
}

File: android/app/src/main/java/io/raindrop/raindropio/nativebridge/NativeBridge.kt
package io.raindrop.raindropio.nativebridge

import android.content.ContentResolver
import android.content.Intent
import android.content.pm.PackageInfo
import android.content.pm.PackageManager
import android.content.res.Configuration
import android.net.Uri
import android.webkit.MimeTypeMap
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

class NativeBridge(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

    private val context: ReactApplicationContext = reactContext

    override fun getName(): String {
        return "NativeBridge"
    }

    override fun getConstants(): Map<String, Any> {
        val constants: MutableMap<String, Any> = HashMap()

        // Is Tablet?
        constants["isTablet"] = false
        try {
            val layout = this.context.resources.configuration.screenLayout and Configuration.SCREENLAYOUT_SIZE_MASK
            constants["isTablet"] = layout == Configuration.SCREENLAYOUT_SIZE_LARGE || layout == Configuration.SCREENLAYOUT_SIZE_XLARGE
        } catch (e: Exception) {
        }

        // Version
        constants["appVersion"] = "0.0.0"
        try {
            val packageManager: PackageManager = this.context.packageManager
            val packageName: String = this.context.packageName
            val info: PackageInfo = packageManager.getPackageInfo(packageName, 0)
            constants["appVersion"] = info.versionName
        } catch (e: Exception) {
        }

        return constants
    }

    @ReactMethod
    fun openFileUrl(url: String, mimeType: String?) {
        val cR: ContentResolver = context.contentResolver
        val mime: MimeTypeMap = MimeTypeMap.getSingleton()

        // Details
        val uri: Uri = Uri.parse(url)
        val detectedMimeType: String? = mime.getExtensionFromMimeType(cR.getType(uri))

        // Start activity
        val intent = Intent(Intent.ACTION_VIEW).apply {
            setDataAndType(uri, mimeType?.takeIf { it.isNotEmpty() } ?: detectedMimeType)
            flags = Intent.FLAG_ACTIVITY_NO_HISTORY
        }
        currentActivity?.startActivity(Intent.createChooser(intent, ""))
    }
}